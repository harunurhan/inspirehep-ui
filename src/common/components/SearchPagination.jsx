import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Pagination } from 'antd';

class SearchPagination extends Component {
  render() {
    const { page, total, pageSize, onPageChange } = this.props;
    return (
      <Pagination
        hideOnSinglePage
        style={{ textAlign: 'center' }}
        current={page}
        onChange={onPageChange}
        total={total}
        pageSize={pageSize}
      />
    );
  }
}

SearchPagination.propTypes = {
  onPageChange: PropTypes.func.isRequired,
  total: PropTypes.number.isRequired,
  page: PropTypes.number,
  pageSize: PropTypes.number,
};

SearchPagination.defaultProps = {
  page: 1,
  pageSize: 25,
};

export default SearchPagination;
