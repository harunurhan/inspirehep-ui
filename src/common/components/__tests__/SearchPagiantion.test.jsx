import React from 'react';
import { shallow } from 'enzyme';

import SearchPagination from '../SearchPagination';

describe('SearchPagination', () => {
  it('renders with all props set', () => {
    const wrapper = shallow(
      <SearchPagination
        total={100}
        onPageChange={jest.fn()}
        page={2}
        pageSize={10}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with only required props set', () => {
    const wrapper = shallow(
      <SearchPagination total={100} onPageChange={jest.fn()} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
