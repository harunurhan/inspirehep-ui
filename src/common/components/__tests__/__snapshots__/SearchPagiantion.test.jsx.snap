// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPagination renders with all props set 1`] = `
<Pagination
  current={2}
  hideOnSinglePage={true}
  onChange={[MockFunction]}
  pageSize={10}
  prefixCls="ant-pagination"
  selectPrefixCls="ant-select"
  style={
    Object {
      "textAlign": "center",
    }
  }
  total={100}
/>
`;

exports[`SearchPagination renders with only required props set 1`] = `
<Pagination
  current={1}
  hideOnSinglePage={true}
  onChange={[MockFunction]}
  pageSize={25}
  prefixCls="ant-pagination"
  selectPrefixCls="ant-select"
  style={
    Object {
      "textAlign": "center",
    }
  }
  total={100}
/>
`;
